R/tmb-methods.R

```diff
@@ -166,33 +166,112 @@ print.mmrm_tmb <- function(x,
                            ...) {
   cat("mmrm fit\n\n")
 
-  n_subjects <- x$tmb_data$n_subjects
-  n_timepoints <- x$tmb_data$n_visits
-  n_obs <- length(x$tmb_data$y_vector)
-  cov_type <- x$formula_parts$corr_type
-  n_theta <- length(x$theta_est)
+  components <- component(x, c(
+    "deviance", "cov_type", "n_theta", "n_subjects",
+    "n_timepoints", "n_obs", "reml", "method",
+    "convergence", "evaluations", "conv_message", "call"
+  ))
 
-  h_print_call(x$call, n_obs, n_subjects, n_timepoints)
-  h_print_cov(cov_type, n_theta)
+  h_print_call(
+    components$call, components$n_obs,
+    components$n_subjects, components$n_timepoints
```

Comment:
e.g. here spend one line for each argument, name arguments and then use `component(x, ...)` to get it.

Reply:
done